File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for h in range(3, 5000//2):
        for w in range(h, 5000 // 2):
            if 2 * (w + h) - 4 != brown:
                continue
            if (w - 2) * (h - 2) == yellow:
                return [w, h]

File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int width, height;
        for(width = 3; width < 5000; width++){
            for(height = 3; height <= width; height++){
                if((width - 2) * (height - 2) == yellow){
                    if(width + height - 2 == brown / 2 ){
                        return new int[]{width, height};
                    }
                }
            }
        }
        return new int[]{};
    }
}